src/Common/MemoryStatisticsOS.cpp

```diff
@@ -103,6 +109,54 @@ MemoryStatisticsOS::Data MemoryStatisticsOS::get() const
     return data;
 }
 
+#endif
+
+#if defined(OS_FREEBSD)
+
+namespace ErrorCodes
+{
+    extern const int CANNOT_SYSCTL;
+    extern const int KERNEL_STRUCTURE_SIZE_MISMATCH;
+}
+
+MemoryStatisticsOS::MemoryStatisticsOS()
+{
+    pagesize = ::getpagesize();
```

Comment:
Can we use `getPageSize` from `base/getPageSize.h` (maybe we need to adjust it for freebsd) ?

Reply:
Yes, works without modifications.